storage_dropbox.go

```diff
@@ -32,37 +54,70 @@ func (backend *StorageDropbox) Description() string {
 	return "Dropbox Storage"
 }
 
-// LoadChunk loads a Chunk from network
+// LoadChunk loads a Chunk from dropbox
 func (backend *StorageDropbox) LoadChunk(shasum string, part, totalParts uint) (*[]byte, error) {
-	return &[]byte{}, ErrChunkNotFound
+	fileName := shasum + "." + strconv.FormatUint(uint64(part), 10) + "_" + strconv.FormatUint(uint64(totalParts), 10)
+
+	obj, _, err := backend.db.Download(filepath.Join(backend.url.Path, "chunks", fileName), "", 0)
+	if err != nil {
+		return nil, err
+	}
+
+	data, err := ioutil.ReadAll(obj)
+	return &data, err
 }
 
-// StoreChunk stores a single Chunk on network
+// StoreChunk stores a single Chunk from dropbox
 func (backend *StorageDropbox) StoreChunk(shasum string, part, totalParts uint, data *[]byte) (size uint64, err error) {
-	return 0, ErrStoreChunkFailed
+	fileName := shasum + "." + strconv.FormatUint(uint64(part), 10) + "_" + strconv.FormatUint(uint64(totalParts), 10)
+
+	if _, err := backend.db.Metadata(fileName, false, false, "", "", 1); err != nil {
+		// Chunk is already stored
+		return 0, nil
+	}
+
+	_, err = backend.db.UploadByChunk(ioutil.NopCloser(bytes.NewReader(*data)), len(*data), filepath.Join(backend.url.Path, "chunks", fileName), true, "")
+	return uint64(len(*data)), err
 }
 
 // LoadSnapshot loads a snapshot
 func (backend *StorageDropbox) LoadSnapshot(id string) ([]byte, error) {
-	return []byte{}, ErrSnapshotNotFound
+	// Getting obj as type io.ReadCloser and reading it out in order to get bytes returned
+	obj, _, err := backend.db.Download(filepath.Join(backend.url.Path, "snapshots", id), "", 0)
+	if err != nil {
+		return nil, err
+	}
+	return ioutil.ReadAll(obj)
 }
 
 // SaveSnapshot stores a snapshot
 func (backend *StorageDropbox) SaveSnapshot(id string, data []byte) error {
+	_, ErrStoreSnapshotFailed := backend.db.UploadByChunk(ioutil.NopCloser(bytes.NewReader(data)), len(data), filepath.Join(backend.url.Path, "snapshots", id), true, "")
 	return ErrStoreSnapshotFailed
 }
 
 // InitRepository creates a new repository
 func (backend *StorageDropbox) InitRepository() error {
+	if _, err := backend.db.CreateFolder(backend.url.Path); err != nil {
+	}
```

Comment:
Should probably check if it's a critical error - or the folder simply already exists.